Indicators/MeanAbsoluteDeviation.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace QuantConnect.Indicators {
+    /// <summary>
+    /// This indicator computes the n-period mean absolute deviation.
+    /// </summary>
+    public class MeanAbsoluteDeviation : WindowIndicator<IndicatorDataPoint> {
+        private readonly IndicatorBase<IndicatorDataPoint> _mean;
```

Comment:
I think this can be a public get private set property so other indicators can use it in composition. Thoughts?